tests/components/media_player/test_async_helpers.py

```diff
@@ -139,121 +138,89 @@ async def async_toggle(self):
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_toggle(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_ON
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_toggle(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_OFF
-
-
-class TestSyncMediaPlayer(unittest.TestCase):
-    """Test the media_player module."""
-
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.player = SyncMediaPlayer(self.hass)
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Shut down test instance."""
-        self.hass.stop()
-
-    def test_volume_up(self):
-        """Test the volume_up helper function."""
-        assert self.player.volume_level == 0
-        self.player.set_volume_level(0.5)
-        assert self.player.volume_level == 0.5
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_volume_up(), self.hass.loop
-        ).result()
-        assert self.player.volume_level == 0.7
-
-    def test_volume_down(self):
-        """Test the volume_down helper function."""
-        assert self.player.volume_level == 0
-        self.player.set_volume_level(0.5)
-        assert self.player.volume_level == 0.5
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_volume_down(), self.hass.loop
-        ).result()
-        assert self.player.volume_level == 0.3
-
-    def test_media_play_pause(self):
-        """Test the media_play_pause helper function."""
-        assert self.player.state == STATE_OFF
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_media_play_pause(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_PLAYING
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_media_play_pause(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_PAUSED
-
-    def test_toggle(self):
-        """Test the toggle helper function."""
-        assert self.player.state == STATE_OFF
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_toggle(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_ON
-        asyncio.run_coroutine_threadsafe(
-            self.player.async_toggle(), self.hass.loop
-        ).result()
-        assert self.player.state == STATE_OFF
+@pytest.fixture
+def async_player(hass):
+    """Fixture of an asynchronous media player."""
+    return AsyncMediaPlayer(hass)
+
+
+@pytest.fixture
+def sync_player(hass):
+    """Fixture of a synchronous media player."""
+    return SyncMediaPlayer(hass)
+
+
+async def test_async_volume_up(hass, async_player):
+    """Test the volume_up helper function."""
+    assert async_player.volume_level == 0
+    await asyncio.ensure_future(
```

Comment:
Could the `asyncio.ensure_future` calls use the new `asyncio.create_task` call instead? It seems like the `ensure_future` eventually `create_task` anyways so calling it directly could save some overhead. I'm new to this though so I could be wrong. The test passes with `await asyncio.create_task(async_player.async_set_volume_level(0.5)`.

Reply:
We  could use `asyncio.create_task` but we don't need to create a task specifically at all. We can just `await async_player.async_set_volume_level` since we're in a coroutine function.

This test is not needed though. It's just testing code that we have defined in this test module. We want to test the code in the media_player integration.

Reply:
That makes sense to me. I just copied the tests and wasn't sure if the event loop had any importance in the tests. You can't specify event loop with a simple `await`, so I went with `ensure_future`.